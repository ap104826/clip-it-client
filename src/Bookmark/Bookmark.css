.Bookmark {
  width: 320px;
  display: flex;
  flex-direction: column;
  align-items: center;
  flex: 0 0 auto;
  width: 18rem;
  max-width: 100%;
  margin-bottom: 1.5rem !important;
  margin-right: 1.5rem !important;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 0.25rem;
}

.Bookmark__img {
  flex-shrink: 0;
  width: 100%;
}

.Bookmark__title {
  font-size: 0.85rem;
  margin-bottom: 0.75rem;
}

.Bookmark__body {
  flex: 1 auto;
  min-height: 1px;
  padding: 1.25rem;
}

.Bookmark__delete {
  --bookmark-button-height: 32px;
  background-color: #041a3a;
  border: 0;
  color: #bdbcbc;
  float: right;
  font-size: 12px;
  height: var(--bookmark-button-height);
  top: calc((var(--bookmark-button-height) + 34px) * -1);
  transform-origin: right bottom;
  margin-right: 5px;
}

.Bookmark__delete:hover {
  background-color: #bd0909;
  color: white;
  cursor: pointer;
  font-weight: bold;
}

.Bookmark__title a {
  color: slategrey;
  display: block;
  text-decoration: none;
}
.Bookmark__title a:hover {
  color: #4ec281;
}

.Bookmark__footer {
  display: flex;
  justify-content: space-between;
  align-items: center;
  min-width: 15rem;
}

.Bookmark__footer-buttons {
  display: flex;
  flex: 2;
  justify-content: flex-end;
}

.Bookmark__category {
  font-size: 14px;
  margin: auto;
  color: #fff;
  background-color: #6c757d;
  display: inline-block;
  padding: 0.25em 0.4em;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: initial;
  border-radius: 0.25rem;
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.Bookmark__fav-icon {
  flex: 1;
}
