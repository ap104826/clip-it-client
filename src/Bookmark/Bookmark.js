import React from 'react'
import { Link } from 'react-router-dom'
import { format } from 'date-fns'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import ApiContext from '../ApiContext'
import { Button, Card } from 'react-bootstrap'
import config from '../config'
import './Bookmark.css'

export default class Bookmark extends React.Component {
  static defaultProps = {
    onDeleteBookmark: () => { },
  }
  static contextType = ApiContext;

  handleClickDelete = e => {
    e.preventDefault()
    const bookmarkId = this.props.id

    // fetch(`${config.API_ENDPOINT}/bookmarks/${bookmarkId}`, {
    //   method: 'DELETE',
    //   headers: {
    //     'content-type': 'application/json'
    //   },
    // })
    //   .then(res => {
    //     if (!res.ok)
    //       return res.json().then(e => Promise.reject(e))
    //     return res.json()
    //   })
    //   .then(() => {
    this.context.deleteBookmark(bookmarkId)
    // allow parent to perform extra behaviour
    // this.props.onDeleteBookmark(bookmarkId)
    // })
    // .catch(error => {
    //   console.error({ error })
    // })
  }

  render() {
    const { title, id, modified, category_id, link, thumbnail_url } = this.props
    return (
      <Card key={id}>
        <Card.Img variant="top" src={`${process.env.PUBLIC_URL}/${thumbnail_url}`} />
        <Card.Body>
          <Card.Title className='Bookmark__title'>
            <a href={link}>{title}</a>
          </Card.Title>
          <Button variant="danger" onClick={this.handleClickDelete}>Delete</Button>
        </Card.Body>
      </Card>
    )
  }
}
