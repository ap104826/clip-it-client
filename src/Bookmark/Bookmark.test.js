import React from 'react'
import ReactDOM from 'react-dom'
import Bookmark from './bookmark'


import { library } from '@fortawesome/fontawesome-svg-core'
import {
  faPlus, faTrash, faChevronLeft, faTrashAlt, faCheckDouble, faBookmark, faShare, faHeart as fasHeart
} from '@fortawesome/free-solid-svg-icons'
import {
  faHeart as farHeart
} from '@fortawesome/free-regular-svg-icons'

library.add(faPlus, fasHeart, farHeart, faChevronLeft, faTrash, faTrashAlt, faCheckDouble, faBookmark, faShare)

describe.only("App renders properly", () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <Bookmark />,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
})
