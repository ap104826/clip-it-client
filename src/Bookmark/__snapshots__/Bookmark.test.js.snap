// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark component renders a .Bookmark by default 1`] = `
<div
  className="Bookmark"
>
  <h2
    className="Bookmark__title"
  >
    <Link
      replace={false}
      to="/bookmark/undefined"
    />
  </h2>
  <button
    className="Bookmark__delete"
    type="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    remove
  </button>
  <div
    className="Bookmark__dates"
  >
    <div
      className="Bookmark__dates-modified"
    >
      Modified
       
      <span
        className="Date"
      >
        Invalid Date
      </span>
    </div>
  </div>
</div>
`;

exports[`Bookmark component renders the Bookmark given props 1`] = `
<div
  className="Bookmark"
>
  <h2
    className="Bookmark__title"
  >
    <Link
      replace={false}
      to="/bookmark/a"
    >
      test-class-name
    </Link>
  </h2>
  <button
    className="Bookmark__delete"
    type="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    remove
  </button>
  <div
    className="Bookmark__dates"
  >
    <div
      className="Bookmark__dates-modified"
    >
      Modified
       
      <span
        className="Date"
      >
        15th Jan 2019
      </span>
    </div>
  </div>
</div>
`;
